### ISSUE TITLE
1.0.8 broke my `error()` derive

### ISSUE THREAD

**Issue by jyn514** [dtolnay/thiserror#53]
I just upgraded from 1.0.4 to 1.0.8 and it broke my build. The following used to compile fine with thiserror:

```rust
use thiserror::Error;

#[derive(Debug)]
struct Inner {
    data: String
}

#[derive(Debug, Error)]
#[error("{}", .0.data)]
struct S(Inner);

fn main() {
    let s = S(Inner { data: "Hello, world!".to_string() });
    println!("{}", s);
}
```

With 1.0.8, I now get an error:

```
error[E0277]: `Inner` doesn't implement `std::fmt::Display`
 --> src/main.rs:9:16
  |
9 | #[error("{}", .0.data)]
  |                ^ `Inner` cannot be formatted with the default formatter
  |
  = help: the trait `std::fmt::Display` is not implemented for `Inner`
  = note: in format strings you may be able to use `{:?}` (or {:#?} for pretty-print) instead
  = note: required by `std::fmt::Display::fmt`

error: aborting due to previous error
```

Is this intentional? If so, why? I don't need to print `.0` to print the error message.

Some more things I tried:

- `#[error("{}", 0.data)]`
```rust
error[E0308]: mismatched types
 --> src/main.rs:9:15
  |
9 | #[error("{}", 0.data)]
  |               ^
  |               |
  |               expected reference, found integer
  |               help: consider borrowing here: `&0`
```

- `#[error("{}", &0.data)]`

```rust
error: expected expression, found `,`
 --> src/main.rs:8:17
  |
8 | #[derive(Debug, Error)]
  |                 ^^^^^ expected expression

error: proc-macro derive produced unparseable tokens
 --> src/main.rs:8:17
  |
8 | #[derive(Debug, Error)]
  |                 ^^^^^
```

- `#[error("{}", (.0).data)]`: `expected reference, found struct 'std::string::String'`
- `#[error("{}", (&.0.data))]` - `error: float literals must have an integer part`
- `#[error("{}", (&self.0.data))]` - works but gives warning about unneccessary parentheses
- `#[error("{}", &self.0.data)]` - `error: proc-macro derive produced unparseable tokens`

**Comment by jyn514** [comment]
Just bisected locally and this first broke in `1.0.7`, `1.0.6` works fine.

**Comment by dtolnay** [comment]
Sorry about the breakage! This was a bug in one of the changes in 1.0.7. I released 1.0.9 with a fix.

**Comment by jyn514** [comment]
Works great! Thanks so much for the quick fix :)
